src/cript/api/rest.py

```diff
@@ -53,7 +53,10 @@ def __init__(self, host: str = None, token: str = None, tls: bool = True):
         }
 
         # Test API authentication by fetching session info
-        response = self.session.get(f"{self.url}/session-info/")
+        try:
+            response = self.session.get(f"{self.url}/session-info/")
+        except Exception as e:
+            raise APIError("Connection API failed, please review your host and token")
```

Comment:
What happens to the `e` exception (line 58) ? The fact you do nothing with it means it will be silent or it will be shown? (I have no knowledge in that)

Reply:
Nothing is really done with it, I had originally intended to use the e to figure out the exact exception but it is a very specific nested exception that can't be captured well, which is why it is handled with the Exception catch all

Reply:
I see, can't you insert `e` in the message into your `APIError` ?

Like that if nobody catch it, it will display:

```
catched error message

your APIError message
```

Reply:
Sorry to bud in here, but I think, this is what you want.
https://www.pythontutorial.net/python-oop/python-raise-from/

That leaves the stack trace intact and its easier to figure out the problem as a user.

Reply:
`raise APIError("Connection API failed, please review your host and token") from e`

Reply:
That's a great comment thank you!